src/sql-parser/src/parser.rs

```diff
@@ -3249,6 +3249,7 @@ impl<'a> Parser<'a> {
         let name = self.parse_item_name()?;
         let columns = self.parse_parenthesized_column_list(Optional)?;
         let in_cluster = self.parse_optional_in_cluster()?;
+        let as_of = self.parse_optional_internal_as_of()?;
```

Comment:
I first had this after the query, to mimic the `SELECT` and `SUBSCRIBE` syntax, but then couldn't get the tests to pass because of an ambiguity issue. That issue has since been fixed in https://github.com/MaterializeInc/materialize/pull/25208, so we could move the `AS OF` after the query again.

Reply:
Moved it after the query again now.